packages/cli/generation/local-generation/local-workspace-runner/src/runLocalGenerationForWorkspace.ts

```diff
@@ -128,26 +137,37 @@ async function writeIrToFile({
     generatorInvocation,
     workspaceTempDir,
     context,
+    irOverrideVersion,
 }: {
     workspace: FernWorkspace;
     audiences: Audiences;
     generatorInvocation: GeneratorInvocation;
     workspaceTempDir: DirectoryResult;
     context: TaskContext;
+    irOverrideVersion?: string;
 }): Promise<AbsoluteFilePath> {
+    const irVersion = irOverrideVersion !== undefined ? irOverrideVersion : "";
+
     const intermediateRepresentation = await generateIntermediateRepresentation({
         workspace,
         audiences,
         generationLanguage: generatorInvocation.language,
     });
-    const migratedIntermediateRepresentation = migrateIntermediateRepresentationForGenerator({
-        intermediateRepresentation,
-        context,
-        targetGenerator: {
-            name: generatorInvocation.name,
-            version: generatorInvocation.version,
-        },
-    });
+    const migratedIntermediateRepresentation =
+        irVersion === ""
+            ? migrateIntermediateRepresentationForGenerator({
+                  intermediateRepresentation,
+                  context,
+                  targetGenerator: {
+                      name: generatorInvocation.name,
+                      version: generatorInvocation.version,
+                  },
+              })
+            : migrateIntermediateRepresentationThroughVersion({
+                  intermediateRepresentation,
+                  context,
+                  version: irVersion,
+              });
```

Comment:
instead of using `""` as a way to figure out if the param was passed, you can setup the ternary like: 

```typescript
irOverrideVersion != null ? 
         migrateIntermediateRepresentationThroughVersion({
                  intermediateRepresentation,
                  context,
                  version: irVersion,
              })
          : migrateIntermediateRepresentationForGenerator({
      intermediateRepresentation,
      context,
      targetGenerator: {
          name: generatorInvocation.name,
          version: generatorInvocation.version,
      },
  });
```